src/RichText/RichText.php

```diff
@@ -346,16 +346,26 @@ private static function fixImagesPath(string $content): string
             "/ (href)='[^']*\/front\/document\.send\.php([^']+)' /",
 
             // src attribute, surrounding by " or '
-            '/ (src)="[^"]*\/front\/document\.send\.php([^"]+)" /',
-            "/ (src)='[^']*\/front\/document\.send\.php([^']+)' /",
+            '/ (src)="(?!data:)[^"]*\/front\/document\.send\.php([^"]+)" /',
+            "/ (src)='(?!data:)[^']*\/front\/document\.send\.php([^']+)' /",
         ];
 
         foreach ($patterns as $pattern) {
-            $content = preg_replace(
+            $result = preg_replace(
                 $pattern,
                 sprintf(' $1="%s/front/document.send.php$2" ', $CFG_GLPI["root_doc"]),
                 $content
             );
+            if ($result === null) {
+                $log_msg =
+                    "preg_replace failed in " . __METHOD__ .
+                    " with pattern: $pattern\n" .
+                    "error: " . preg_last_error_msg() . "\n" .
+                    "content length: " . strlen($content);
```

Comment:
The stack trace will be already available in logs, the message could be simplified and should not contain newlines, whenever it is possible.

The content length will not help as the limits preg operations depends on the PHP configuration/version.

```suggestion
                $log_msg =sprintf(
                    '`preg_replace()` with pattern `%s` failed: `%s`.',
                    $pattern,
                    preg_last_error_msg()
                );
```